src/js/recipe.spec.js

```diff
@@ -0,0 +1,84 @@
+		before(() => {
+			recipe = new Recipe(1, ingredients, steps, "mexicana", "china", 1);
+		});
+		it("El id debería de devolver 1", () => {
+			expect(recipe.id).to.equals(1);
+		});
+		it("El ingrediente deberia ser igual a \"'1,' leche\"", () => {
+			expect(recipe.ingredients).to.equals(ingredients);
+		});
+		it("Los pasos deberían ser \"'a','b','c','d','f','g'\"", () => {
+			expect(recipe.steps).to.equals(steps);
+		});
+		it("El tipo de comida deberia ser igual a 'mexicana'", () => {
+			expect(recipe.typeOfFood).to.equals("mexicana");
+		});
+		it("El origen debería ser 'china'", () => {
+			expect(recipe.origin).to.equals("china");
+		});
+		it("El id del autor debería ser 1", () => {
+			expect(recipe.idAuthor).to.equals(1);
+		});
+	});
+	describe("Prueba de excepciones", () => {
+		it("Deberia saltar una exepcion si el id no es un numero", () => {
+			const idError = () => new Recipe("", ingredients, steps, "mexicana", "china", 1);
+			expect(idError).to.throw();
+		});
+		it("Debería saltar una excepcion si el id es negativo", () => {
+			const idError2 = () => new Recipe(-1, ingredients, steps, "mexicana", "china", 1);
+			expect(idError2).to.throw();
+		});
+		it("Debería saltar una excepcion si ingredientes no es un array", () => {
+			const ingredientsError = () => new Recipe(1, "ingrediente", steps, "mexicana", "china", 1);
+			expect(ingredientsError).to.throw();
+		});
+		it("Debería saltar una excepcion si el array de ingredients es menor a 3", () => {
+			const ingredientsError2 = () => new Recipe(1, ingredients.splice(-1, 1), steps, "mexicana", "china", 1);
+			expect(ingredientsError2).to.throw();
+		});
+		it("Debería saltar una excepcion si el contenido del array ingredients no son Ingredients", () => {
+			const ingredientsError3 = () => new Recipe(1, steps, steps, "mexicana", "china", 1);
+			expect(ingredientsError3).to.throw();
+		});
+		it("Debería saltar una excepcion si el array steps es menor a 5", () => {
+			const stepsError = () => new Recipe(1, ingredients, steps.splice(-1, 1), "mexicana", "china", 1);
+			expect(stepsError).to.throw();
+		});
+		it("Debería saltar una excepcion si el contenido del array steps no son una cadena", () => {
+			const stepsError2 = () => new Recipe(1, ingredients, ingredients, "mexicana", "china", 1);
+			expect(stepsError2).to.throw();
+		});
+		it("Debería saltar una excepcion si el typeOfFood no es una cadena", () => {
+			const typeOfFoodError = () => new Recipe(1, ingredients, steps, 1, "china", 1);
+			expect(typeOfFoodError).to.throw();
+		});
+		it("Debería saltar una excepcion si Origin no es una cadena", () => {
+			const originError = () => new Recipe(1, ingredients, steps, "mexicana", 1, 1);
+			expect(originError).to.throw();
+		});
+		it("Debería saltar una excepcion si el idAuthor es negativo", () => {
+			const idAuthorError = () => new Recipe(1, ingredients, steps, "mexicana", "china", -1);
+			expect(idAuthorError()).to.throw();
```

Comment:
Este test falla porque estás llamando a `idAuthorError`.  Cuando pasas `idAuthorError`, estás pasando a `expect` una referencia a la función. `expect` la ejecutará y comprobará si hay un error. Cuando pasas `idAuthorError()` estás ejecutando la función y pasando el valor que devuelve a `expect`, en este caso `undefined` porque lanza una excepción y por lo tanto no devuelve nada.

La excepción se lanza, pero en el contexto de `it`, no en el contexto de `expect`, por lo tanto el test falla.